app/src/main/java/me/guillaumin/android/osmtracker/activity/TrackDetail.java

```diff
@@ -119,6 +138,11 @@ protected void onResume() {
 			finish();
 			return;  // <--- Early return ---
 		}
+		
+		if (trackData == null)
```

Comment:
Yeah I think that's unnecessary. If you want to be sure, just initialise the field at the top with an empty map instead.